### ISSUE TITLE
Gopacket fails to properly decode SNAP layers

### ISSUE THREAD

**Issue by Galadros** [google/gopacket#613]
Gopacket fails to decode SNAP layers when using a parser structured like the one in `parser.go`  It tries to parse two LLC layers in a row and then tries to parse an unknown layer type, failing.

The reason that this happens is that the `CanDecode()` method of the SNAP struct in `llc.go` mistakenly returns `LayerTypeLLC`.  It should be returning `LayerTypeSNAP`- almost all of the `CanDecode()` methods I've found return their own layer type (e.g. `LLC.CanDecode()` returns `LayerTypeLLC`, `Dot11MgmtProbeReq.CanDecode()` returns `LayerTypeDot11MgmtProbeReq`, etc.)  As-is, gopacket treats the bytes following the LLC layer as a _second_ LLC layer.  When the `LLC.NextLayerType()` (`llc.go` lines 61-69) function runs, it doesn't find a correct value:
```
func (l *LLC) NextLayerType() gopacket.LayerType {
	switch {
	case l.DSAP == 0xAA && l.SSAP == 0xAA:
		return LayerTypeSNAP
	case l.DSAP == 0x42 && l.SSAP == 0x42:
		return LayerTypeSTP
	}
	return gopacket.LayerTypeZero // Not implemented
}
```

Since gopacket is now trying to treat a section of bytes as an LLC when they _aren't_ in fact an LLC, `l.DSAP` and `l.SSAP` refer to arbitrary bytes, which won't have one of those special values mentioned.  `NextLayerType()` then returns `LayerTypeZero`, which errors as an unknown layer type, replicating the above symptoms.

The incorrect code, on lines 97-100 of `llc.go`:
```
// CanDecode returns the set of layer types that this DecodingLayer can decode.
func (s *SNAP) CanDecode() gopacket.LayerClass {
	return LayerTypeLLC
}
```

What it should be:
```
// CanDecode returns the set of layer types that this DecodingLayer can decode.
func (s *SNAP) CanDecode() gopacket.LayerClass {
	return LayerTypeSNAP
}
```

**Comment by notti** [comment]
Whops that was a copy & paste error on my side.
Can you move this from #542 into it's own pull request, so we can merge it?

**Comment by Galadros** [comment]
@notti yeah I'll do so soon.  The problem is that my master branch on my fork from gopacket is pretty dirty from the changes I've made in #542, and I can't figure out how to get a clean fork from the current HEAD that gopacket is at.  Do you have any suggestions?

**Comment by notti** [comment]
Move your changes to a branch (it's always easier in github to work on branches and make a pull request from there).
Then you can also rebase your branch to enjoy the current master changes.

**Comment by Galadros** [comment]
@notti will do, thanks.  I realized a while ago to always make changes on branches, but to be honest I was originally intending that fork to be very short so I got sloppy.  Whoops.

If I already committed some changes to master, is there any way to go back and peel them off into a branch of their own?  I really appreciate your help on this, by the way- I'm much less skilled with Git than I'd like to be.

**Comment by notti** [comment]
Hmm I think adding a new branch and then moving master back with git reset should work. Commits can be moved aroud /edited with interactive rebase (https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)
Although if you do this, your pull request (it was from master) will be affected (you would need to create a new one).
Maybe you could also rebase your master onto the current gopacket master, and move the one commit to it's own branch.
